west.yml

```diff
@@ -0,0 +1,38 @@
+# SOF west manifest
+manifest:
+  defaults:
+    remote: thesofproject
+
+  remotes:
+    - name: thesofproject
+      url-base: https://github.com/thesofproject
+    - name: zephyrproject
+      url-base: https://github.com/zephyrproject-rtos
+
+  projects:
+    - name: rimage
+      repo-path: rimage
+      path: sof/rimage
```

Comment:
Please try to use a different location. Removing submodules _and_ replacing them by something else in the same place makes switching branches super painful, breaks `git bisect` etc.

Reply:
I'm looking into it. However now I restored rimage as git submodule and it should be ok.
If west clones to very same directory where submodule is expected, clones very same revision of repo then git during command:
`git submodule update --init --recursive`
gives you only information of registering:
![image](https://user-images.githubusercontent.com/12482515/179041689-72245b51-67e1-4c5d-ae19-32d47fc5baa6.png)

Reply:
Ok I see a potencial bad scenario:

1. `west update` clones rimage and tomlc99 but does not registers them as submodules on branch A
2. end-user does not runs `git submodule update --init --recursive` (because why would he/she)?
3. End-user checks out to branch B that have submodules registered to other SHA than branch A.
4. End-user runs platform build that fails with eg. image sining
5. End-user runs git status that does not show any changes because submodules had not been registered at all (and they are actually there cloned by west and checked out to wrong revision).

However:

1. Note, that that's how west revision control works. It does not show git submodules change because its not git.
The fact that we do not remove git submodules is temporary because we can't do it easily.
2. I don't think that cloning "submodules" using `west update` to let's say new _rimage_west_ directory is beneficial. Yes you would avoid confusion described in bad scenario above but the cost of redoing scripts, headers and adding tons of logic everywhere "it not rimage dir use rimage_west dir". Seems not worth it to me.